common/flatpak-dir.c

```diff
@@ -1486,6 +1538,50 @@ flatpak_dir_system_helper_call_ensure_repo (FlatpakDir   *self,
   return ret != NULL;
 }
 
+static gboolean
+flatpak_dir_system_helper_call_get_revokefs_fd (FlatpakDir   *self,
+                                                guint         arg_flags,
+                                                const gchar  *arg_ref,
+                                                const gchar  *arg_commit,
+                                                const gchar  *arg_installation,
+                                                gint         *out_socket,
+                                                gchar       **out_src_dir,
+                                                GCancellable *cancellable,
+                                                GError      **error)
+{
+  g_autoptr(GUnixFDList) out_fd_list = NULL;
+  gint fd = -1;
+  gint fd_index = -1;
+
+  if (flatpak_dir_get_no_interaction (self))
+    arg_flags |= FLATPAK_HELPER_GET_REVOKEFS_FD_FLAGS_NO_INTERACTION;
+
+  g_debug ("Calling system helper: GetRevokefsFd");
+
+  g_autoptr(GVariant) ret =
+    flatpak_dir_system_helper_call (self, "GetRevokefsFd",
+                                    g_variant_new ("(usss)",
+                                                   arg_flags,
+                                                   arg_ref,
+                                                   arg_commit,
+                                                   arg_installation),
+                                    G_VARIANT_TYPE ("(hs)"),
+                                    &out_fd_list,
+                                    cancellable, error);
+
+  if (ret == NULL)
+    return FALSE;
+
+  g_variant_get (ret,
+                 "(hs)",
+                 &fd_index,
+                 out_src_dir);
+  fd  = g_unix_fd_list_get (out_fd_list, fd_index, NULL);
```

Comment:
This needs to check that fd != -1 and not pass NULL for error, because this is actually a dup() call that can fail if you're e.g. out of fds.

Reply:
Addressed and introduced error handling